miso-web/src/spring-test/java/uk/ac/bbsrc/tgac/miso/webapp/springtest/UserRestControllerST.java

```diff
@@ -0,0 +1,129 @@
+package uk.ac.bbsrc.tgac.miso.webapp.springtest;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.Arrays;
+
+import javax.ws.rs.core.MediaType;
+
+import org.junit.Test;
+import org.springframework.security.test.context.support.WithMockUser;
+
+import uk.ac.bbsrc.tgac.miso.core.data.impl.UserImpl;
+import uk.ac.bbsrc.tgac.miso.dto.Dtos;
+import uk.ac.bbsrc.tgac.miso.dto.UserDto;
+import uk.ac.bbsrc.tgac.miso.webapp.controller.rest.UserRestController.PasswordChangeDto;
+import static org.springframework.test.web.servlet.setup.MockMvcBuilders.*;
+import javax.ws.rs.core.MediaType;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.*;
+
+
+public class UserRestControllerST extends AbstractST {
+
+
+  private static final String CONTROLLER_BASE = "/rest/users";
+  private static final Class<UserImpl> entityClass = UserImpl.class;
+
+
+  // create, update, delete, search, change password
+  private UserDto makeCreateDto() throws Exception {
+    UserDto dto = new UserDto();
+    dto.setAdmin(false);
+    dto.setActive(true);
+    dto.setInternal(true);
+    dto.setLoginName("new");
+    dto.setFullName("new guy");
+    dto.setPassword("OntarioInstituteOfCancerResearch2025");
+    return dto;
+  }
+
+  @Test
+  @WithMockUser(username = "admin", password = "admin", roles = {"INTERNAL", "ADMIN"})
+  public void testCreate() throws Exception {
+    // must be admin to create user
+    UserImpl user = baseTestCreate(CONTROLLER_BASE, makeCreateDto(), entityClass, 200);
+    assertEquals(false, user.isAdmin());
+    assertEquals(true, user.isActive());
+    assertEquals(true, user.isInternal());
+    assertEquals("new", user.getLoginName());
+    assertEquals("new guy", user.getFullName());
```

Comment:
If you save the results of makeCreateDto earlier in this function, you could replace these hard-coded expected values with the actual values of the result of makeCreateDto (e.g. `assertEquals(dto.isAdmin(), user.isAdmin())` )